src/hooks/useStripeProductsAndCurrency.ts

```diff
@@ -0,0 +1,84 @@
+import { currencyService } from '@/components/services/currency.service';
+import { ProductsProps, stripeService } from '@/components/services/stripe.service';
+import { useEffect, useReducer } from 'react';
+import { CouponType } from '@/pages/api/stripe/get_coupons';
+
+interface UseStripeProductsAndCurrencyResponse {
+  products: ProductsProps | undefined;
+  loadingCards: boolean;
+  currency: { symbol: string; value: number };
+  coupon: CouponType | null;
+}
+
+type ActionType =
+  | { type: 'SET_PRODUCTS'; payload: ProductsProps | undefined }
+  | { type: 'SET_LOADING_CARDS'; payload: boolean }
+  | { type: 'SET_CURRENCY'; payload: { symbol: string; value: number } }
+  | { type: 'SET_COUPON'; payload: CouponType | null };
+
+const reducer = (state: any, action: ActionType) => {
+  switch (action.type) {
+    case 'SET_PRODUCTS':
+      return { ...state, products: action.payload };
+    case 'SET_LOADING_CARDS':
+      return { ...state, loadingCards: action.payload };
+    case 'SET_CURRENCY':
+      return { ...state, currency: action.payload };
+    case 'SET_COUPON':
+      return { ...state, coupon: action.payload };
+    default:
+      return state;
+  }
+};
+
+function useStripeProductsAndCurrency(couponCode?: CouponType): UseStripeProductsAndCurrencyResponse {
```

Comment:
I'm not a fan of mixing product/provider names (Stripe) with actual code, cause if we change the payments platform in the future, this will not make sense anymore.

My recommendation: `usePricing`, which is provider agnostic, and more "generic" style, so it could work as a "dump box" for pricing stuff